File: recyclerview/src/main/java/com/melayer/recyclerview/MainActivity.java
package com.melayer.recyclerview;

import androidx.appcompat.app.AppCompatActivity;
import androidx.lifecycle.ViewModelProvider;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

import android.os.Bundle;
import android.view.ViewManager;

import java.util.ArrayList;
import java.util.List;

public class MainActivity extends AppCompatActivity {


    private RecyclerViewModel vmp;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        vmp = new ViewModelProvider(this).get(RecyclerViewModel.class);

//        List<WaData> list = new ArrayList<WaData>();
//        list.add(
//                new WaData(
//                        R.drawable.ic_android_black_24dp,
//                        "Android",
//                        "Hey Hi",
//                        "7.45 Am",
//                        3,
//                        true
//                )
//        );
//
//        list.add(
//                new WaData(
//                        R.drawable.ic_launcher_background,
//                        "Volume",
//                        "its okay",
//                        "8.00 Pm",
//                        45,
//                        false
//                )
//        );

        LinearLayoutManager mgr = new LinearLayoutManager(this);
        WaAdapter adapter = new WaAdapter(this, null);

        RecyclerView recLst = findViewById(R.id.waList);
        recLst.setLayoutManager(mgr);
        recLst.setAdapter(adapter);
    }
}

File: app/src/main/java/com/melayer/onlineapp/MainActivity.kt
package com.melayer.onlineapp

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.util.Log
import androidx.databinding.DataBindingUtil
import androidx.lifecycle.Observer
import androidx.lifecycle.ViewModelProvider
import com.melayer.onlineapp.databinding.ActivityMainBinding

class MainActivity : AppCompatActivity() {

    private val viewModel by lazy {
        ViewModelProvider(this)
            .get(CalcData::class.java)
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

       DataBindingUtil.setContentView<ActivityMainBinding>(this, R.layout.activity_main).apply {
           viewmodel = viewModel
           lifecycleOwner = this@MainActivity
       }

        viewModel.num1.observe(this, Observer {
            Log.i("@ani", "Changed Num1 $it")
            val num1 = if(it.isNotEmpty()) Integer.parseInt(it ?: "0") else 0
            val num2 = Integer.parseInt(viewModel.num2.value ?: "0")
            viewModel.setResult(num1 + num2)
        })

        viewModel.num2.observe(this, Observer {
            Log.i("@ani", "Changed Num1 $it")
            val num1 = Integer.parseInt(viewModel.num1.value ?: "0")
            val num2 = if(it.isNotEmpty()) Integer.parseInt(it ?: "0") else 0
            viewModel.setResult(num1+num2)
        })
    }
}